frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -365,6 +365,12 @@
             onSuccess: function(response) {
                 var details = response.data;
                 vm.phaseSplits = details;
+                for(var i=0; i<details.length; i++) {
+                    if (details[i].visibility !== 3) {
```

Comment:
Checking directly with number can be confusing at times. Can you please check with JS object? 

Something like:

```
var challengePhaseVisibility = {
    PUBLIC: 3,
    PRIVATE: 2,
    ABC: 1
}
if details[i].visibility == challengePhaseVisibility.PUBLIC
```